docs/TA_properties.md

```diff
@@ -0,0 +1,134 @@
+`TA_FLAGS` by **user_ta_header_defines.h**.
+
+## 1.1. Single Instance Property
+
+`"gpd.ta.singleInstance"` is a boolean property of the TA.
+
+This property defines if one instance of the TA must be created and will
+receive all open session request, or if a new specific TA instance must be
+created for each incoming open session request.
+
+OP-TEE TA flag `TA_FLAG_SINGLE_INSTANCE` sets to configuration of this
+property.
+
+The boolean property is set to `true` if `TA_FLAGS` sets bit
+`TA_FLAG_SINGLE_INSTANCE`, otherwise the boolean property is set to `false`.
+
+## 1.2. Multi-session Property
+
+`"gpd.ta.multiSession"` is a boolean property of the TA.
+
+This property defines if the TA instance can handle reveral sessions of if any
+open session request is proceed only no there is no pending opened session
+towardds the TA. This property is useless if TA is NOT SingleInstance.
+
+OP-TEE TA flag `TA_FLAG_MULTI_SESSION` sets to configuration of this
+property.
+
+The boolean property is set to `true` if `TA_FLAGS` sets bit
+`TA_FLAG_MULTI_SESSION`, otherwise the boolean property is set to `false`.
+
+## 1.3. Keep Alive Property
+
+`"gpd.ta.instanceKeepAlive"` is a boolean property of the TA.
+
+This property defines if the TA instance created must be destroyed or not when
+all sessions opened towards the TA are closed. It the property is enabled, TA
+instance, once created (at 1st open session request), is never removed unless
+the TEE itsefl is restarted (boot/reboot).
+
+OP-TEE TA flag `TA_FLAG_INSTANCE_KEEP_ALIVE` sets to configuration of this
+property.
+
+The boolean property is set to `true` if `TA_FLAGS` sets bit
+`TA_FLAG_INSTANCE_KEEP_ALIVE`, otherwise the boolean property is set to `false`.
+
+## 1.4. Heap Size Property
+
+`"gpd.ta.dataSize"` is a 32bit interger property of the TA.
+
+This property defines the size in bytes of the TA allocation pool, in which
+`TEE_Malloc()` and friends allocate memory.
+
+The value of the property must be defined by the macro `TA_DATA_SIZE` from
+**user_ta_header_defines.h**.
+
+## 1.5. Stack Size Property
+
+`"gpd.ta.stackSize"` is a 32bit interger property of the TA.
+
+This property defines the size in bytes of the stack used for TA execution.
+
+The value of the property must be defined by the macro `TA_STACK_SIZE` from
+**user_ta_header_defines.h**.
+
+# 2. TA Property Flags from the OP-TEE Extensions
+
+## 2.1. User Mode Property Flag
+
+`TA_FLAG_USER_MODE` is a bit flag supported by `TA_FLAGS`.
+
+This property flag is currently meaningless in OP-TEE. It may be set or not
+without impact on TA execution. All OP-TEE TAs are executed in user mode/level.
+
+Therefore any can set or not set `TA_FLAG_USER_MODE` in their `TA_FLAGS`
+without impact. We recommend to discard enable and use of this property flag.
+
+## 2.2. Exec-in-DDR Property Flag
+
+`TA_FLAG_EXEC_DDR` is a bit flag supported by `TA_FLAGS`.
+
+This property flag is currently meaningless in OP-TEE. Nevertheless it shall
```

Comment:
Nothing wrong with the text, but it sounds a bit strange to say something is meaningless, but still should be set. Is it GP tests that checks this flag? Otherwise I think we should simply remove it from OP-TEE.

Reply:
yes, we should remove the flag.